test/integration/api/OrderAPI.spec.ts

```diff
@@ -894,6 +894,49 @@ describe('OrderAPI', () => {
         );
       });
     });
+
+    describe('when liquidity amounts don’t match amount of components of issuance order', async () => {
+      beforeEach(async () => {
+        const takerWalletOrder = subjectOrders[0] as TakerWalletOrder;
+        takerWalletOrder.takerTokenAmount = takerWalletOrder.takerTokenAmount.sub(1);
+        subjectOrders[0] = takerWalletOrder;
+      });
+
+      test('throws', async () => {
+        const takerWalletOrder = subjectOrders[0] as TakerWalletOrder;
+
+        const {
+          takerTokenAddress,
+          takerTokenAmount,
+        } = takerWalletOrder;
+
+        const {
+          requiredComponentAmounts,
+          requiredComponents,
+          quantity,
+        } = subjectSignedIssuanceOrder;
+
+        let takerIndex = 0;
+        _.each(requiredComponents, (component, i) => {
+          if (component === takerTokenAddress) {
+            takerIndex = i;
+          }
+        });
+
+        const desiredComponentFillAmount =
+          calculatePartialAmount(
+            requiredComponentAmounts[takerIndex],
+            subjectQuantityToFill,
+            quantity,
+          );
+
+        return expect(subject()).to.be.rejectedWith(
+          `Token amount of ${takerTokenAddress} from liquidity sources, ${takerTokenAmount.toString()}, ` +
```

Comment:
We need to converge upon stringifying BigNumbers or not. Other parts of our codebase are not doing this. We need to be consistent. I'll create a card.

Reply:
i just know that `toNumber()` sometimes does some rounding which we have to be careful about I think which is why i did toString() here